### PR TITLE
feat(jellyfish-api-core): add blockchain getMempoolEntry RPC

### PR DESCRIPTION
**Pull request by JustinWong98**
#### What this PR does / why we need it:

/kind feature

#### Which issue(s) does this PR fixes?:
Implemented type for RPC: https://github.com/DeFiCh/jellyfish/issues/48
-  blockchain.getMempoolEntry()

### CODE REVIEW

File: packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should getMempoolEntry and return MempoolTx', async () => {
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts:21]: try test with invalid txid

**Comment by fuxingloh** [comment]
image


And retitle your PR to follow: https://github.com/DeFiCh/jellyfish/blob/main/.github/labeler.yml

File: packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    await waitForExpect(async () => {
+      const txId = await client.wallet.sendToAddress('mwsZw8nF7pKxWH8eoKL9tPxTpaFkz7QeLU', 0.002)
+      const mempoolEntry: MempoolTx = await client.blockchain.getMempoolEntry(txId)
+      expect(mempoolEntry.vsize instanceof BigNumber).toStrictEqual(true)
```
Comment (ivan-zynesis) [inline packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts:25]: Minor syntax suggestion, thinking 
```
expect(mempoolEntry).toStrictEqual({
  vsize: expect.any(BigNumber)
  ...
})
```
will be cleaner

File: packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        spentby: expect.any(Array),
+        'bip125-replaceable': expect.any(Boolean)
+      })
+    }, 10000)
+  })
+
+  it('should return error for invalid txId', async () => {
+    const invalidTxIdErr = /RpcApiError: 'Transaction not in mempool', code: -5, method: getmempoolentry/
+    await expect(client.blockchain.getMempoolEntry('6b1bac73bf8071e7edecef30081058f342ff35be12eb2dd3aa1d2ec4933ee798'))
+      .rejects.toThrow(invalidTxIdErr)
```
Comment (eli-lim) [inline packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts:50]: Can simplify by inlining the error msg
```suggestion
    await expect(client.blockchain.getMempoolEntry('6b1bac73bf8071e7edecef30081058f342ff35be12eb2dd3aa1d2ec4933ee798'))
      .rejects.toThrow(`RpcApiError: 'Transaction not in mempool', code: -5, method: getmempoolentry`)
```

File: packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  })
+
+  it('should return error for txid of invalid length', async () => {
+    const invalidtxidstring = "RpcApiError: 'parameter 1 must be of length 64 (not 17, for 'invalidtxidstring')', code: -8, method: getmempoolentry"
+    await expect(client.blockchain.getMempoolEntry('invalidtxidstring')).rejects.toThrow(invalidtxidstring)
```
Comment (eli-lim) [inline packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts:55]: Can simplify by inlining the error msg

```suggestion
    await expect(client.blockchain.getMempoolEntry('invalidtxidstring'))
      .rejects.toThrow(`RpcApiError: 'parameter 1 must be of length 64 (not 17, for 'invalidtxidstring')', code: -8, method: getmempoolentry`)
```

File: packages/jellyfish-api-core/src/category/blockchain.ts [hunk 5]
```diff
@@ -164,6 +164,15 @@ export class Blockchain {
     return await this.client.call('getrawmempool', [verbose], 'bignumber')
   }
 
+  /**
+   *
+   * @param {string} txId the transaction id
```
Comment (eli-lim) [inline packages/jellyfish-api-core/src/category/blockchain.ts:169]: Add description aligned with https://github.com/DeFiCh/ain/blob/63452763fef6d4dc7adf9df8e371def8ecd80caf/src/rpc/blockchain.cpp#L711

```suggestion
   * Get mempool data for the given transaction
   * @param {string} txId the transaction id
```

File: docs/node/CATEGORIES/01-blockchain.md [hunk 6]
```diff
@@ -237,6 +237,42 @@ interface blockchain {
...
+    'bip125-replaceable': boolean
+  }
+}
+```
+## getMempoolEntry
+
+Get transaction details in the memory pool using a transaction ID.
+
+```ts title="client.blockchain.getMempoolEntry(txId: string)"
 interface MempoolTx {
```
Comment (eli-lim) [inline docs/node/CATEGORIES/01-blockchain.md:276]: `MempoolTx` interface is already defined above. No need to redefine it.
Instead, add interface for calling

```typescript
interface blockchain {
  getMempoolEntry (txId: string): Promise<MempoolTx>
}
```

**Review by eli-lim — CHANGES_REQUESTED** [review]

**Review by ivan-zynesis — APPROVED** [review]

**Review by eli-lim — APPROVED** [review]

**Review by canonbrother — APPROVED** [review]

File: packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import BigNumber from 'bignumber.js'
+import { MempoolTx } from '../../../src/category/blockchain'
+import waitForExpect from 'wait-for-expect'
+
+describe('getMempoolInfo', () => {
```
Comment (cwkang1998) [inline packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts:7]:
```suggestion
describe('getMempoolEntry', () => {
```

**Review by cwkang1998 — DISMISSED** [review]

File: packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts (hunk 7 above)
Comment (JustinWong98) [reply packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolEntry.test.ts]: Ah, thanks for pointing this out!

**Review by eli-lim — APPROVED** [review]

**Review by canonbrother — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**